src/dramatiq_abort/middleware.py

```diff
@@ -144,53 +144,74 @@ def abort(
         message_id: str,
         abort_ttl: Optional[int] = None,
         mode: AbortMode = AbortMode.ABORT,
+        abort_timeout: int = 0,
     ) -> None:
         if abort_ttl is None:
             abort_ttl = self.abort_ttl
-        modes = [AbortMode.CANCEL]
+        modes: List[Tuple[AbortMode, Dict[str, Any]]]
+        modes = [(AbortMode.CANCEL, dict())]
         if mode is AbortMode.ABORT:
-            modes.append(AbortMode.ABORT)
-        keys = [self.id_to_key(message_id, mode) for mode in modes]
-        self.backend.notify(keys, ttl=abort_ttl)
+            modes.append((AbortMode.ABORT, dict(abort_timeout=abort_timeout)))
+        values = [(self.id_to_key(message_id, mode), args) for mode, args in modes]
+        self.backend.notify(values, ttl=abort_ttl)
+
+    def check_abort_request(self) -> Optional[float]:
+        abort_time: Optional[float]
+        _, abort_time = self.manager.abort_requests.get(
+            threading.get_ident(), (None, None)
+        )
+        if abort_time is None:
+            return None
+        return 1000 * (abort_time - time.monotonic())
 
-    def _handle(self) -> None:
-        message_ids = list(self.manager.abortables.keys())
+    def _get_abort_requests(self) -> None:
+        message_ids = self.manager.abortable_messages.keys()
```

Comment:
why did you remove the `list`? It's required to avoid iterating on mutated .dictionnary

Reply:
Nice catch! I had edited that code and forgot the list when I changed it back